nexus_constructor/instrument_view/q_components.py

```diff
@@ -0,0 +1,168 @@
+from typing import List
+
+import numpy as np
+from PySide2.Qt3DCore import Qt3DCore
+from PySide2.Qt3DExtras import Qt3DExtras
+from PySide2.Qt3DRender import Qt3DRender
+from PySide2.QtGui import QColor, QMatrix4x4, QVector3D
+
+from nexus_constructor.instrument_view.off_renderer import OffMesh
+from nexus_constructor.instrument_view.qentity_utils import (
+    MATERIAL_ALPHA,
+    MATERIAL_COLORS,
+    MATERIAL_DIFFUSE_COLORS,
+    create_material,
+    create_qentity,
+)
+
+
+class QComponent:
+    def __init__(self, root_entity: Qt3DCore.QEntity, nx_class: str):
+        self.root_entity = root_entity
+        self.nx_class = nx_class
+
+        self.entities: List[Qt3DCore.QEntity] = []
+        self.default_material: Qt3DRender.QMaterial = self._create_default_material()
+
+    def create_entities(self):
+        raise NotImplementedError
+
+    def add_transformation(self, transformation: Qt3DCore.QComponent):
+        raise NotImplementedError
+
+    def remove_transformation(self, transformation: Qt3DCore.QComponent):
+        for entity in self.entities:
+            entity.removeComponent(transformation)
+
+    def setParent(self, value=None):
+        for entity in self.entities:
+            entity.setParent(value)
+
+    def get_entity(self):
```

Comment:
I assume there is a reason why it returns the first element in the list? Not super clear to me. Maybe rename the function or add docstring explaining it.